openlibrary/plugins/openlibrary/js/SearchBar.js

```diff
@@ -0,0 +1,298 @@
+            this.$input.val(SearchBar.marshalBookSearchQuery(q));
+        }
+        this.$form.attr('action', this.composeSearchUrl(this.$input.val()));
+        SearchUtils.addModeInputsToForm(this.$form, SearchUtils.mode.read());
+    }
+
+    /** Initialize event handlers that allow the form to collapse for small screens */
+    initCollapsibleMode() {
+        this.toggleCollapsibleModeForSmallScreens($(window).width());
+        $(window).resize(debounce(() => {
+            this.toggleCollapsibleModeForSmallScreens($(window).width());
+        }, 50));
+        $(document).on('submit','.in-collapsible-mode', event => {
+            if (this.collapsed) {
+                event.preventDefault();
+                this.toggleCollapse();
+                this.$input.focus();
+            }
+        });
+    }
+
+    /**
+     * Enables/disables CollapsibleMode depending on screen size
+     * @param {Number} windowWidth
+     */
+    toggleCollapsibleModeForSmallScreens(windowWidth) {
+        if (windowWidth < 568) {
+            if (!this.inCollapsibleMode) {
+                this.enableCollapisbleMode();
+                this.collapse();
+            }
+            this.clearAutocompletionResults();
+        } else {
+            if (this.inCollapsibleMode) {
+                this.disableCollapsibleMode();
+            }
+        }
+    }
+
+    /** Collapses or expands the searchbar */
+    toggleCollapse() {
+        if (this.collapsed) {
+            this.expand();
+        } else {
+            this.collapse();
+        }
+    }
+
+    collapse() {
+        $('header#header-bar .logo-component').removeClass('hidden');
+        this.$component.removeClass('expanded');
+        this.collapsed = true;
+    }
+
+    expand() {
+        $('header#header-bar .logo-component').addClass('hidden');
+        this.$component.addClass('expanded');
+        this.collapsed = false;
+    }
+
+    enableCollapisbleMode() {
+        this.$form.addClass('in-collapsible-mode');
+        this.inCollapsibleMode = true;
+    }
+
+    disableCollapsibleMode() {
+        this.collapse();
+        this.$form.removeClass('in-collapsible-mode');
+        this.inCollapsibleMode = false;
+    }
+
+    /**
+     * Converts an already processed query into a search url
+     * @param {String} q query that's ready to get passed to the search endpoint
+     * @param {Boolean} [json] whether to hit the JSON endpoint
+     * @param {Number} [limit] how many items to get
+     */
+    composeSearchUrl(q, json, limit) {
+        let url = this.facetEndpoint;
+        if (json) {
+            url += '.json';
+        }
+        url += `?q=${q}`;
+        if (limit) {
+            url += `&limit=${limit}`;
+        }
+        url += `&mode=${SearchUtils.mode.read()}`;
+        return url;
+    }
+
+    /**
+     * Prepare an unprocessed query for book searching
+     * @param {String} q
+     * @return {String}
+     */
+    static marshalBookSearchQuery(q) {
```

Comment:
Why not make it live outside the class since it's static?

Reply:
WONTFIX as benefits are minimal if any